segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java

```diff
@@ -2054,6 +2064,194 @@ public void testSnapshotInfo() throws Exception {
         container.stopAsync().awaitTerminated();
     }
 
+    /**
+     * Test that the {@link ContainerEventProcessor} service is started as part of the {@link StreamSegmentContainer}
+     * and that it can process events.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testBasicContainerEventProcessor() throws Exception {
+        @Cleanup
+        TestContext context = createContext();
+        val container = (StreamSegmentContainer) context.container;
+        container.startAsync().awaitRunning();
+        ReusableLatch latch = new ReusableLatch();
+        final AtomicReference<String> userEvent = new AtomicReference<>("event1");
+        Function<List<BufferView>, CompletableFuture<Void>> handler = l -> {
+            l.forEach(s -> Assert.assertEquals(userEvent.get().length(), s.getLength()));
+            return CompletableFuture.runAsync(latch::release);
+        };
+        @Cleanup
+        ContainerEventProcessorImpl containerEventProcessor = new ContainerEventProcessorImpl(container, container.metadataStore,
+                TIMEOUT_EVENT_PROCESSOR_ITERATION, TIMEOUT_EVENT_PROCESSOR_ITERATION, this.executorService());
+        ContainerEventProcessor.EventProcessorConfig eventProcessorConfig =
+                new ContainerEventProcessor.EventProcessorConfig(EVENT_PROCESSOR_EVENTS_AT_ONCE, EVENT_PROCESSOR_MAX_OUTSTANDING_BYTES);
+        @Cleanup
+        ContainerEventProcessor.EventProcessor processor = containerEventProcessor.forConsumer("testConsumer", handler, eventProcessorConfig)
+                .get(TIMEOUT_FUTURE.toSeconds(), TimeUnit.SECONDS);
+
+        // Test adding one Event to the EventProcessor and wait for the handle to get executed and unblock this thread.
+        BufferView event = new ByteArraySegment(userEvent.get().getBytes());
+        processor.add(event, TIMEOUT_FUTURE).join();
+        latch.await();
+        latch.reset();
+
+        // Test the same with a larger events.
+        userEvent.set("longerEvent2");
+        event = new ByteArraySegment(userEvent.get().getBytes());
+        processor.add(event, TIMEOUT_FUTURE).join();
+        processor.add(event, TIMEOUT_FUTURE).join();
+        processor.add(event, TIMEOUT_FUTURE).join();
+        latch.await();
+
+        // Check that if the same EventProcessor name is used, the same object is retrieved.
+        Assert.assertEquals(processor, containerEventProcessor.forConsumer("testConsumer", handler, eventProcessorConfig)
+                .get(TIMEOUT_FUTURE.toSeconds(), TimeUnit.SECONDS));
+    }
+
+    /**
+     * Test the behavior of the {@link StreamSegmentContainer} when multiple {@link ContainerEventProcessor.EventProcessor}
+     * objects are registered, including one with a faulty handler.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testMultipleEventProcessors() throws Exception {
```

Comment:
It will return the same object on all the calls, because the implementation is synchronized. In any case, I have added a test called `testConcurrentForConsumerCall()` for this.